ban-cooking.lua

```diff
@@ -0,0 +1,440 @@
+
+        for r, g in ipairs(p.growths) do
+            matinfo = dfhack.matinfo.decode(g)
+            m = matinfo.material
+
+            if m.flags["EDIBLE_COOKED"] and m.flags["LEAF_MAT"] then
+                for j, s in ipairs(p.material) do
+                    if m.id == s.id then
+
+                        -- need to create a key to reference values in the types_list dictionary.
+                        key_matinfo_type = matinfo.type .. i
+
+                        if not(type_list[key_matinfo_type]) then
+                            type_list[key_matinfo_type] = {}
+                        else
+                            print('Key exists for ' .. p.name, type_list[key_matinfo_type]["text"], type_list[key_matinfo_type]["type"], j + matinfo.type)
+                        end
+
+                        type_list[key_matinfo_type]["text"] = p.name .. " " .. m.id .. " growth"
+                        -- item type for plant materials listed in the growths array appear to always use the 'PLANT_GROWTH' item type tag
+                        type_list[key_matinfo_type]["type"] = "PLANT_GROWTH"
+                        -- item subtype is equal to the array index of the cookable item in the growths table
+                        type_list[key_matinfo_type]["subtype"] = r
+                    end
+                end
+            end
+        end
+    end
+    -- cycle through all creatures
+    for i, c in ipairs(df.global.world.raws.creatures.all) do
+        for j, m in ipairs(c.material) do
+            -- need to create a key to reference values in the types_list dictionary.
+            key_matinfo_type = j + matinfo.type .. i
+
+            -- print(key_matinfo_type[1])
+            -- print(key_matinfo_type[2])
+
+            if m.reaction_product and m.reaction_product.id and reaction_product_id_contains(m.reaction_product, "CHEESE_MAT") then
+                if not(type_list[key_matinfo_type]) then
+                    type_list[key_matinfo_type] = {}
+                end
+
+                type_list[key_matinfo_type]["text"] = c.name[1] .. " milk"
+                -- item type for milk appears to use the 'LIQUID_MISC' tag
+                type_list[key_matinfo_type]["type"] = "LIQUID_MISC"
+                type_list[key_matinfo_type]["subtype"] = -1
+
+            end
+
+            if m.reaction_product and m.reaction_product.id and reaction_product_id_contains(m.reaction_product, "SOAP_MAT") then
+                if not(type_list[key_matinfo_type]) then
+                    type_list[key_matinfo_type] = {}
+                end
+
+                type_list[key_matinfo_type]["text"] = c.name[1] .. " tallow"
+                -- item type for milk appears to use the 'GLOB' tag
+                type_list[key_matinfo_type]["type"] = "GLOB"
+                type_list[key_matinfo_type]["subtype"] = -1
+            end
+        end
+    end
+
+    iterator = 0
+
+    output = {}
+
+    for i, b in pairs(already_banned) do
```

Comment:
`already_banned` cannot be persisted in global state since it would get out of sync with manual "unbanning" actions taken by the player. therefore this will only work to show items that were *just* banned in the current invocation of the script. This is fine, but I wanted to make sure you were aware of the implications. You'd have to use it like `ban-cooking fruit tallow show`. just running `ban-cooking show` by itself would show nothing.

Reply:
Maybe I already fixed this issue with some of my other changes, but I just tested using ban-cooking all before saving and fully exiting out of df, restarting it, reloading the save, and calling ban-cooking show with the output being equal to before I restarted df. Also I manually banned and unbanned some items in my kitchen interface in df and the changes show up on running ban-cooking show. Am I missing something here?